cellxgene_resources/curation_qa.ipynb

```diff
@@ -455,19 +450,22 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "prop = 'sample_id'\n",
+    "prop = 'donor_id'\n",
     "\n",
     "results = []\n",
     "for a in adata.obs[prop].unique():\n",
-    "    print(a)\n",
     "    obs_df = adata.obs[adata.obs[prop] == a]\n",
     "    num_to_check = obs_df.shape[0] // 5 #default is check 20% of barcodes\n",
     "    r = TENx_barcode_checker(obs_df, num_to_check)\n",
-    "    if r:\n",
-    "        r[prop] = a\n",
-    "        results.append(r)\n",
-    "    else:\n",
-    "        print('no barcodes checked')\n",
+    "    print(r)\n",
+    "    r_dict = r['summary'].value_counts().to_dict()\n",
+    "    r_dict[prop] = a\n",
+    "    r_list = ['3pv2_5pv1_5pv2', '3pv3', 'multiome', 'multiple', 'none']\n",
```

Comment:
I think this and the following for loop are unnecessary.
You can just append `.fillna(0)` to the last line so you don't have NaNs clogging up the report

Reply:
The last line as in: `pd.DataFrame(results).set_index(prop).fillna(0)`? Not sure how you want results formatted. Just output of TENx_barcode_checker()?

Reply:
Correct. That last line.
Full thought process...
The `r_list = `... and following loop lines take 4 lines of code to fill in 0s where you don't have any data.
If you remove those, then the last line with the DataFrame will fill in those gaps with NaNs.
Probably easier on the eyes to have those NaNs replaced with 0s, but you can take care of that with just appending `.fillna(0)` as you've noted.
So same read-out, but ditch 4 lines of code.

Reply:
and add `astype(int)` to the final line, or else you'll get everything on the table as `.0`
So last line is `pd.DataFrame(results).set_index(prop).fillna(0).astype(int)`